EVI5L.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+from typing import Union
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+import numpy as np
+from abc import ABC, abstractmethod
+
+
+def check(args: list[str]):
+    """
+    Checks list of sequences for correct format.
+    Arguments:
+        -args (list) - Input format: at least 1 sequence and command, command on last position of list,
+        only DNA/RNA seqeunces.
+    Return:
+        -ValueError - if something does not fit arguments input format.
+    """
+    if len(args) < 2:
+        raise ValueError('Input sequences and command')
+    elif args[-1] not in COMMANDS:
+        raise ValueError('No such command')
+    for seq in args[:-1]:
+        if set(seq.lower()) == set('autgc') and set(seq.lower()) != set('atgc') and set(seq.lower()) != set('augc'):
+            raise ValueError('Incorrect nucleic acid')
+
+
+def filter_fastq(
+        input_path: str,
+        output_filename: str = None,
+        gc_bounds: Union[int, tuple[int, int]] = (0, 100),
+        length_bounds: Union[int, tuple[int, int]] = (0, 2 ** 32),
+        quality_treshholds: int = 0
+) -> dict[str, str]:
+    """
+    Filters fastq files by specifiable parameters.
+    Arguments:
+        -input_path(str) - path to input file.
+        -output_filename (str) - name of output file, input file name will be used if none is given.
+        -gc_bounds (int,tuple) - GC interval (in percent) for filtering, default is (0, 100). If input is single int - it will be ceiling (0, n).
+        -length_bounds (int,tuple) - length interval for filtering. Works exactly as gc_bounds. Default is (0, 2**32)
+        -quality_threshold (int) - The threshold value of average read quality for the filter is 0 by default (phred33 scale).
+    Return:
+        -output (dict[str,str]) - filtered dictionary, which consists of entities that fulfill entered parameters.
+    """
+
+    def average_quality(record):
+        return np.mean(record.letter_annotations["phred_quality"])
+
+    if output_filename is None:
+        output_filename = input_path
+
+    with open(input_path, "r") as input_handle, open(output_filename, "w") as output_handle:
+        sequences = SeqIO.parse(input_handle, "fastq")
+        filtered = {}
+        for seq in sequences:
+            if len(seq) >= length_bounds[0] and len(seq) <= length_bounds[1] \
+                    and average_quality(seq) >= quality_threshold \
+                    and GC(seq.seq) >= gc_bounds[0] and GC(seq.seq) <= gc_bounds[1]:
```

Comment:
Я бы постаралась вычислить GC состав один раз:
```suggestion
gc_content = GC(seq.seq)
if len(seq) >= length_bounds[0] and len(seq) <= length_bounds[1] \
        and average_quality(seq) >= quality_threshold \
        and gc_bounds[0] <= gc_content <= gc_bounds[1]:
```